processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -480,6 +485,20 @@ public ColumnCapabilities getColumnCapabilities(String columnName)
     return null;
   }
 
+  @Nullable
+  public ColumnCapabilities getColumnHandlerCapabilities(String columnName)
+  {
+    if (timeAndMetricsColumnCapabilities.containsKey(columnName)) {
+      return timeAndMetricsColumnCapabilities.get(columnName);
+    }
+    synchronized (dimensionDescs) {
+      if (dimensionDescs.containsKey(columnName)) {
+        return dimensionDescs.get(columnName).getIndexer().getHandlerCapabilities();
+      }
+    }
```

Comment:
Why the `containsKey` then `get`?  Why not just `get()` and check for null?  `containsKey` is doing the exact same work that `get` does, so this is just duplicating work.